lib/rules/no-restricted-class.js

```diff
@@ -0,0 +1,221 @@
+        node
+      ) {
+        const classes = node.value.expression.properties
+          .filter(
+            (property) =>
+              property.type === 'Property' &&
+              (property.key.type === 'Literal' ||
+                property.key.type === 'Identifier')
+          )
+          .reduce((acc, property) => {
+            // Ugly early return to make TS happy
+            if (
+              property.type !== 'Property' ||
+              (property.key.type !== 'Literal' &&
+                property.key.type !== 'Identifier')
+            )
+              return acc
+            let value = ''
+            if (property.key.type === 'Literal') {
+              value = (property.key.value || '').toString()
+            } else {
+              value = property.key.name
+            }
+            const values = value.split(/\s+/)
+            values.forEach((/** @type {string} */ className) => {
+              acc.add(className)
+            })
+            return acc
+          }, new Set())
+        reportForbiddenClasses(classes, node, context, forbiddenClasses)
+      },
+
+      // <Element :class="`forbidden ${someVar}`" />
+      /**
+       * @param {VAttribute & { value: { expression: TemplateLiteral } } } node
+       */
+      'VAttribute[directive=true][key.argument.name="class"][value.expression.type="TemplateLiteral"]'(
+        node
+      ) {
+        const strings = node.value.expression.quasis.reduce(
+          (acc, templateElement) => {
+            const classNames = templateElement.value.raw.split(/\s+/)
+            classNames.forEach((className) => {
+              acc.add(className)
+            })
+            return acc
+          },
+          new Set()
+        )
+        reportForbiddenClasses(strings, node, context, forbiddenClasses)
+      },
+
+      // <Element :class="'forbidden'" />
+      /**
+       * @param {VAttribute & { value: { expression: Literal } } } node
+       */
+      'VAttribute[directive=true][key.argument.name="class"][value.expression.type="Literal"]'(
+        node
+      ) {
+        if (!node.value.expression.value) return
+        const classNames = new Set(
+          node.value.expression.value.toString().split(/\s+/)
+        )
+        reportForbiddenClasses(classNames, node, context, forbiddenClasses)
+      },
+
+      // <Element :class="'forbidden' + ' otherClass'" />
+      /**
+       * @param {VAttribute & { value: { expression: BinaryExpression } } } node
+       */
+      'VAttribute[directive=true][key.argument.name="class"][value.expression.type="BinaryExpression"]'(
+        node
+      ) {
+        if (node.value.expression.left.type === 'PrivateIdentifier') return
+        const classNames = new Set(
+          flattenBinaryExpression(
+            node.value.expression.left,
+            node.value.expression.right
+          ).split(/\s+/)
+        )
+        reportForbiddenClasses(classNames, node, context, forbiddenClasses)
+      }
+    })
```

Comment:
Array-Syntax also needs to be processed.

https://vuejs.org/v2/guide/class-and-style.html#Array-Syntax